Minio/DataModel/BucketOperationsArgs.cs

```diff
@@ -0,0 +1,156 @@
+            this.BucketName = bucket;
+            return this;
+        }
+
+        public new BucketExistsArgs WithLocation(string loc)
+        {
+            this.Location = loc;
+            return this;
+        }
+        public BucketExistsArgs WithRegion(string reg)
+        {
+            this.Region = reg;
+            return this;
+        }
+        public new RestRequest GetRequest(string baseUrl, RestSharp.Authenticators.IAuthenticator authenticator)
+        {
+            return RequestUtil.CreateRequest(baseUrl, Method.HEAD, authenticator, this.BucketName, this.Secure, this.Region );
+        }
+    }
+
+    public class RemoveBucketArgs : BucketArgs
+    {
+        public new void Validate()
+        {
+            utils.ValidateBucketName(this.BucketName);
+        }
+        public new RemoveBucketArgs WithBucket(string bucket)
+        {
+            this.BucketName = bucket;
+            return this;
+        }
+
+        public new RemoveBucketArgs WithLocation(string loc)
+        {
+            this.Location = loc;
+            return this;
+        }
+        public RemoveBucketArgs WithRegion(string reg)
+        {
+            this.Region = reg;
+            return this;
+        }
+        public new RestRequest GetRequest(string baseUrl, RestSharp.Authenticators.IAuthenticator authenticator)
+        {
+            return RequestUtil.CreateRequest(baseUrl, Method.DELETE, authenticator, this.BucketName, this.Secure, this.Region);
+        }
+
+    }
+
+    public class ListObjectsArgs : BucketArgs
+    {
+        public new void Validate()
+        {
+            utils.ValidateBucketName(this.BucketName);
+        }
+        public new ListObjectsArgs WithLocation(string loc)
+        {
+            this.Location = loc;
+            return this;
+        }
+        public ListObjectsArgs WithRegion(string reg)
+        {
+            this.Region = reg;
+            return this;
+        }
+        public new RestRequest GetRequest(string baseUrl, RestSharp.Authenticators.IAuthenticator authenticator)
+        {
+            return RequestUtil.CreateRequest(baseUrl, Method.DELETE, authenticator, this.BucketName, this.Secure,  this.Region);
+        }
+
+        public new IObservable<Item> ProcessResponse(RestSharp.IRestResponse response)
+        {
+            //TODO: Process Items
+            return null;
+        }
+    }
+
+    public class MakeBucketArgs : BucketArgs
+    {
+        public new void Validate()
+        {
+            utils.ValidateBucketName(this.BucketName);
+        }
+
+        public MakeBucketArgs WithBucket(string bucket)
+        {
+            this.BucketName = bucket;
+            return this;
+        }
+
+        public new MakeBucketArgs WithLocation(string loc)
+        {
+            this.Location = loc;
+            return this;
+        }
+        public MakeBucketArgs WithRegion(string reg)
+        {
+            this.Region = reg;
+            return this;
+        }
```

Comment:
remove above methods as they are inherited.

Reply:
Changed to call base method with `new`.